spec/services/invoices/calculate_fees_service_spec.rb

```diff
@@ -356,49 +365,427 @@
+
+              expect(result).to be_success
+              expect(invoice.fees.fixed_charge.count).to eq(0)
+            end
+          end
+
+          context "when changing the subscription" do
+            let(:add_on1) { create(:add_on, organization:) }
+            let(:add_on2) { create(:add_on, organization:) }
+
+            let(:old_plan) { create(:plan, organization:, interval:, pay_in_advance:, trial_period:) }
+            let(:new_plan) { create(:plan, organization:, interval:, pay_in_advance:, trial_period:, amount_cents: 2000) }
+
+            let(:old_fixed_charge) do
+              create(
+                :fixed_charge,
+                plan: old_plan,
+                add_on: add_on1,
+                charge_model: "standard",
+                pay_in_advance: true,
+                properties: {amount: "10"}
+              )
+            end
+
+            let(:new_fixed_charge1) do
+              create(
+                :fixed_charge,
+                plan: new_plan,
+                add_on: add_on1,
+                charge_model: "standard",
+                pay_in_advance: true,
+                properties: {amount: "10"}
+              )
+            end
+
+            let(:new_fixed_charge2) do
+              create(
+                :fixed_charge,
+                plan: new_plan,
+                add_on: add_on2,
+                charge_model: "standard",
+                pay_in_advance: true,
+                properties: {amount: "15"}
+              )
+            end
+
+            let(:old_subscription) do
+              create(
+                :subscription,
+                plan: old_plan,
+                organization:,
+                customer:,
+                billing_time:,
+                subscription_at: started_at,
+                started_at:,
+                created_at:,
+                status: :terminated,
+                terminated_at: timestamp - 1.day
+              )
+            end
+
+            let(:new_subscription) do
+              create(
+                :subscription,
+                plan: new_plan,
+                organization:,
+                customer:,
+                billing_time:,
+                subscription_at: started_at,
+                started_at: timestamp,
+                created_at: timestamp,
+                status: :active,
+                previous_subscription: old_subscription
+              )
+            end
+
+            let(:new_fixed_charge1_event) do
+              create(:fixed_charge_event, fixed_charge: new_fixed_charge1, subscription: new_subscription, timestamp: timestamp, units: 1)
+            end
+
+            let(:new_fixed_charge2_event) do
+              create(:fixed_charge_event, fixed_charge: new_fixed_charge2, subscription: new_subscription, timestamp: timestamp, units: 1)
+            end
+
+            before do
+              old_fixed_charge
+              fixed_charge.update(pay_in_advance: false)
```

Comment:
why not redefine the fixed charge with the config we would like to have?

Here `update` will silently fail and continue with the test hiding real behaviour, if you actually want to update use `update!` instead but I would highly recommend redefining the variable with the right config.

```
let(:fixed_charge) do
  create(:fixed_charge, .... pay_in_advance: false)
end
```

even `pay_in_advance` is already false by default... hence no need to be explicit here.